yt_dlp/postprocessor/sponsorblock.py

```diff
@@ -0,0 +1,391 @@
+                if not real_download and not move_to_sponsored.warned:
+                    move_to_sponsored.warned = True
+                    self.report_warning('Removing SponsorBlock segments multiple times may cut out '
+                                        'unintended parts of the video')
+                os.rename(file, orig_file_enc)
+            to_remove.append(orig_file)
+            return orig_file
+        move_to_sponsored.warned = False
+
+        orig_video = move_to_sponsored(video_file)
+        self._remove_segments(orig_video, video_file, duration, sponsor_chapters)
+
+        # get(..., {}) is broken: sometimes a literal None is stored under this key.
+        subs = info.get('requested_subtitles')
+        if not subs:
+            return to_remove, info
+
+        for lang, sub in subs.items():
+            ext = sub['ext']
+            if ext not in self.SUPPORTED_SUBS:
+                self.report_warning(
+                    'Cannot remove SponsorBlock segments from external subtitles '
+                    'of type ".%s". They are now out of sync.' % ext)
+                continue
+
+            sub_file = sub['filepath']
+            orig_sub = move_to_sponsored(sub_file)
+            self._remove_segments(orig_sub, sub_file, duration, sponsor_chapters)
+
+        return to_remove, info
+
+    def _get_segments_plaintext(self, video_id):
+        url = '%s/api/skipSegments?%s' % (self._api, compat_urllib_parse_urlencode({
+            'videoID': video_id,
+            'categories': json.dumps(self._categories),
+            'service': 'YouTube'
+        }))
+        return self._get_json(url)
+
+    def _get_segments_private(self, video_id):
+        hash = sha256(video_id.encode('ascii')).hexdigest()
+        # SponsorBlock API recommends using first 4 hash characters.
+        url = '%s/api/skipSegments/%s?%s' % (self._api, hash[:4], compat_urllib_parse_urlencode({
+            'categories': json.dumps(self._categories),
+            'service': 'YouTube'
+        }))
+        for d in self._get_json(url):
+            if d['videoID'] == video_id:
+                return d['segments']
+        return []
+
+    def _get_json(self, url):
+        def make_request(url):
+            try:
+                self.write_debug('SponsorBlock query: %s' % url)
+                rsp = self._downloader.urlopen(sanitized_Request(url))
+                return json.loads(rsp.read())
+            except compat_HTTPError as e:
+                if e.code == 404:
+                    return []
+                raise PostProcessingError('Error communicating with SponsorBlock API')
+
+        if re.match('^https?://', url):
+            return make_request(url)
+        else:  # Custom API endpoint without a protocol. Try HTTPS then fallback to insecure.
+            try:
+                return make_request('https://' + url)
+            except PostProcessingError:
+                return make_request('http://' + url)
```

Comment:
You should hard error when you can't request over HTTPS and ask the user to specify HTTP if that's what they want, I actually just realised I overlooked this in Sponskrub but you shouldn't do the same because it's bad security practice.

Reply:
OK, I'll remove the `http` branch.